### ISSUE TITLE
Investigate cherry picking automation

### ISSUE THREAD

**Issue by GCHQ-Developer-530** [mi6/ic-ui-kit#2669]
## Summary
At the moment we're merging work into develop and then cherry picking the commits into V3 develop. There's quite a bit of human error here though as we might miss some commits going into V3. It would be good if there was a way of automating it.

## 💬 Description 
Look into whether we can have a github workflow that:
- Gets triggered after a merge into develop
- Gets any new commit ids
- Cherry picks them into a new branch off V3
- Opens a PR from the new branch to V3 develop

Also make the same workflow on the design system repo too.

## Why do we need it? 
Reduce the likelihood of skipping commits when cherry picking.

**Comment by evenstensberg** [comment]
Can I fix this?

**Comment by GCHQ-Developer-530** [comment]
Hi @evenstensberg, yeah of course, go for it! As long as it's all standard action work and no third party usage, then that'd be great

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Okay, great. Why don't you just git rebase develop v3 from the develop branch instead of cherry-picking?

**Comment by GCHQ-Developer-530** [comment]
We previously had an action doing that, however there were always too many merge conflicts and we'd have to do it manually. We noticed that a couple of times work went missing by us doing it manually, so cherry picking is the best way moving forward.

**Comment by evenstensberg** [comment]
Ok.

# Resources

- https://stackoverflow.com/questions/11835948/git-cherry-pick-vs-rebase
- https://github.com/gorillio/github-action-cherry-pick
- https://github.com/marketplace/actions/github-cherry-pick-action

**Comment by evenstensberg** [comment]
I'll fix this after #2617 is merged.

**Comment by GCHQ-Developer-530** [comment]
We need to get #2617 checked by someone internally who is currently on leave so that PR is on hold for a bit. We'd like to get this one resolved as soon as possible as it's slowing the team down.

**Comment by ad9242** [comment]
Currently having some issues with permissions trying to push the newly created branch: eg: https://github.com/mi6/ic-ui-kit/actions/runs/11701936408/job/32589576253

moved back to "ready for dev" for further investigation

**Comment by evenstensberg** [comment]
@ad9242 I can't change any permissions myself, but I can find articles/links to what needs to be adjusted in the GitHub settings(?)

**Comment by ad9242** [comment]
the action seems to squash multiple commits from the initial PR into a single one, so we currently need to find a way to maintain all the original commit

**Comment by evenstensberg** [comment]
I’ve read the docs and it doesn’t support cherry picking each commit from a pr. I’ll see if there’s some other actions that supports that
